### PR TITLE
Adding a new docs-only release

### PR DESCRIPTION
**Pull request by dozoisch**
For #552 

This is a work in progress, I just wanted some feedback about this!

@mtscout6 if this is the same thing as what you had in mind, I'll go ahead and do the two other points!

STEPS:
- [x] update Maintaining about cherry-picking to "new" docs branch
- [x] extract small script for doc release that tags with tag-docs.X

### CODE REVIEW

File: tools/release-docs-scripts/pre-conditions.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        throw new Error(`Must be on ${docBranchName} branch to do docs-only releases. Currently: ${currentBranchName}`);
+      }
+      console.info(`On ${docBranchName} branch`.cyan);
+    });
+}
+
+export default function docsReleasePreConditions() {
+  return releasePreConditions().then(ensureBranchDocs);
+}
+
```
Comment (mtscout6) [inline tools/release-docs-scripts/pre-conditions.js]: I don't think we need to enforce the name of the branch. This part can just be a documentation point.

File: tools/release-scripts/constants.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const docsRoot = path.join(repoRoot, 'docs-built/');
+
+const tmpBowerRepo = path.join(repoRoot, 'tmp-bower-repo');
+const tmpDocsRepo = path.join(repoRoot, 'tmp-docs-repo');
+
+export {
+  repoRoot,
+  bowerRepo, bowerRoot, tmpBowerRepo,
+  docsRoot, docsRepo, tmpDocsRepo,
+};
```
Comment (mtscout6) [inline tools/release-scripts/constants.js]: Nice I like this cleanup of repeated code!

File: tools/release-docs-scripts/pre-conditions.js (hunk 1 above)
Comment (mtscout6) [inline tools/release-docs-scripts/pre-conditions.js]: I do agree on the same preconditions as the mainstream release process though :+1:

**Comment by mtscout6** [comment]
One of your checkpoints:

> Make sure master gets merged in docs when a release is made.

If the documentation updates are cherry-picked from `master` there's no real need to merge them to `master`. Or is this your thought around having a dedicated `docs` branch name? Where the mainstream release makes sure that the `docs` branch is merge before an actual release ensues.

**Comment by dozoisch** [comment]
Actually what I meant is that we need to merge master in docs after a release to make sure we get the docs branch on par with master.

**Comment by mtscout6** [comment]
Sounds like you are thinking about having a long lived `docs` branch?

**Comment by dozoisch** [comment]
Yeah that what I was thinking, or maybe we can create a new docs branch when creating a new release (at the first cherrypick)? Like docs/0.22 we can probably automate the branch creation. Like your are on master you do tools/docs-release <hash> and the tool auto checkout or create docs/version , cherry picks the commit, tag and push

**Comment by mtscout6** [comment]
I'm thinking this is overkill. For example consider a pull request to master containing multiple commits that all need to applied to a docs release.

The docs release does not need to be so rigid as to control branches for us, that is a minor manual control to sling commits around by cherry-picking or even an interactive rebase.

**Comment by dozoisch** [comment]
Alright!

**Comment by dozoisch** [comment]
@mtscout6  Alright, I removed the precondition for the branch name and updated MAINTAINING.md, let me know what you think about that!

**Comment by mtscout6** [comment]
This looks fantastic! Thanks for putting this together.

**Comment by mtscout6** [comment]
@dozoisch Did you have anything else you wanted to do with this? If not you can go ahead and merge it in.

**Comment by dozoisch** [comment]
@mtscout6  Nope not really, I'll squash and reword then merge.

**Comment by mtscout6** [comment]
:+1:

**Pull request merged**